### PR TITLE
support export and inference pix2struct model with openvino

### PR DESCRIPTION
**Pull request by eaidova**
# What does this PR do?
Add support OVModelForPix2Struct inference and fix openvino export for models that required preprocessor for getting export configuration

## Before submitting
- [ ] This PR fixes a typo or improves the docs (you can dismiss the other checks if that's the case).
- [x] Did you make sure to update the documentation with your changes?
- [x] Did you write any new necessary tests?

### CODE REVIEW

File: optimum/intel/openvino/modeling_seq2seq.py [hunk 1]
```diff
@@ -125,6 +125,56 @@
...
+    Example of pix2struct:
+
+    ```python
+    >>> from transformers import {processor_class}
+    >>> from optimum.onnxruntime import {model_class}
+    >>> from PIL import Image
+    >>> import requests
+
+    >>> processor = {processor_class}.from_pretrained("{checkpoint}")
+    >>> model = {model_class}.from_pretrained("{checkpoint}", export=True, use_io_binding=True)
```
Comment (eaidova) [inline optimum/intel/openvino/modeling_seq2seq.py:166]:
```suggestion
    >>> model = {model_class}.from_pretrained("{checkpoint}", export=True)
```

File: optimum/intel/openvino/modeling_seq2seq.py [hunk 2]
```diff
@@ -125,6 +125,56 @@
...
+            `(batch_size, num_heads, encoder_sequence_length, embed_size_per_head)`.
+"""
+_PROCESSOR_FOR_DOC = "AutoProcessor"
+
+PIX2STRUCT_EXAMPLE = r"""
+    Example of pix2struct:
+
+    ```python
+    >>> from transformers import {processor_class}
+    >>> from optimum.onnxruntime import {model_class}
```
Comment (eaidova) [inline optimum/intel/openvino/modeling_seq2seq.py:161]:
```suggestion
    >>> from optimum.intel import {model_class}
```

**Comment by eaidova** [comment]
@AlexKoff88 @helena-intel @echarlaix  could you please take a look?

File: optimum/intel/openvino/modeling_seq2seq.py [hunk 3]
```diff
@@ -310,26 +366,33 @@ class OVEncoder:
...
         # Model inputs
-        inputs = {"input_ids": input_ids}
+        if input_ids is None and self.main_input_name == "input_ids":
+            raise RuntimeError("input_ids should be provided")
+        elif self.main_input_name != "input_ids":
+            if self.main_input_name not in kwargs:
+                raise RuntimeError(f"{self.main_input_name} should be provided")
+            inputs = {self.main_input_name: kwargs[self.main_input_name]}
+        else:
+            inputs = {"input_ids": input_ids}
```
Comment (echarlaix) [inline optimum/intel/openvino/modeling_seq2seq.py:395]: could make sense to have :
```suggestion
        inputs = {self.main_input_name: input_ids}
```

File: optimum/intel/openvino/modeling_seq2seq.py [hunk 4]
```diff
@@ -444,3 +511,107 @@ def _compile(self):
...
+        decoder_attention_mask: Optional[torch.BoolTensor] = None,
+        encoder_outputs: Optional[Tuple[Tuple[torch.Tensor]]] = None,
+        past_key_values: Optional[Tuple[Tuple[torch.Tensor]]] = None,
+        **kwargs,
+    ) -> Seq2SeqLMOutput:
+        # Encode if needed : first prediction pass
+        # Encode if needed (training, first prediction pass)
+        if encoder_outputs is None:
+            encoder_outputs = self.encoder(
+                flattened_patches=flattened_patches,
```
Comment (echarlaix) [inline optimum/intel/openvino/modeling_seq2seq.py:579]:
```suggestion
                input_ids=flattened_patches,
```

(for compatibility with suggestion above)

File: optimum/intel/__init__.py [hunk 5]
```diff
@@ -94,6 +94,7 @@
             "OVModelForFeatureExtraction",
             "OVModelForImageClassification",
             "OVModelForMaskedLM",
+            "OVModelForPix2Struct",
```
Comment (echarlaix) [inline optimum/intel/__init__.py:97]: can you also add it here https://github.com/huggingface/optimum-intel/blob/ce6c6bc3bba2de86aa33be8da62290de94d9e938/optimum/intel/__init__.py#L188

Comment (echarlaix) [reply optimum/intel/__init__.py:97]: Could you also add it here as well 



https://github.com/huggingface/optimum-intel/blob/ce6c6bc3bba2de86aa33be8da62290de94d9e938/optimum/intel/openvino/utils.py#L87


https://github.com/huggingface/optimum-intel/blob/ce6c6bc3bba2de86aa33be8da62290de94d9e938/optimum/intel/utils/dummy_openvino_objects.py#L106

**Review by echarlaix — COMMENTED** [review]
thanks for the addition @eaidova

File: optimum/intel/__init__.py (hunk 5 above)
Comment (eaidova) [reply optimum/intel/__init__.py:97]: Updated, thanks

File: optimum/intel/openvino/modeling_seq2seq.py (hunk 3 above)
Comment (eaidova) [reply optimum/intel/openvino/modeling_seq2seq.py]: looks like it does not work in generate, where encoder called outside of common forward

**Review by echarlaix — APPROVED** [review]

**Pull request merged**